hoot-core/src/main/cpp/hoot/core/OsmMap.cpp

```diff
@@ -933,6 +883,13 @@ bool OsmMap::validate(bool strict) const
 
 void OsmMap::visitRo(ElementVisitor& visitor) const
 {
+  visitNodesRo(visitor);
+  visitWaysRo(visitor);
+  visitRelationsRo(visitor);
+}
+
+void OsmMap::visitNodesRo(ElementVisitor& visitor) const
```

Comment:
visit*Ro - I'm feeling there's a way to collapse this down to a single method with templates or something....but I'm going to guess you already thought about something like that, and its not possible given the underlying element code :-)  and its only three methods anyway...